zenodo/modules/metrics/api.py

```diff
@@ -0,0 +1,76 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of Zenodo.
+# Copyright (C) 2016, 2017 CERN.
+#
+# Zenodo is free software; you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 2 of the
+# License, or (at your option) any later version.
+#
+# Zenodo is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Zenodo; if not, write to the
+# Free Software Foundation, Inc., 59 Temple Place, Suite 330, Boston,
+# MA 02111-1307, USA.
+#
+# In applying this license, CERN does not
+# waive the privileges and immunities granted to it by virtue of its status
+# as an Intergovernmental Organization or submit itself to any jurisdiction.
+
+"""Zenodo Metrics API."""
+
+from __future__ import absolute_import
+
+from flask import current_app
+from invenio_accounts.models import User
+from invenio_communities.models import Community
+from invenio_files_rest.models import FileInstance
+
+
+class ZenodoMetric(object):
+    """API class for Zenodo Metrics."""
+
+    metrics_start_date = current_app.config['ZENODO_METRICS_START_DATE']
```

Comment:
from werkzeug.locals import LocalProxy

```suggestion
    metrics_start_date = LocalProxy(lambda: current_app.config['ZENODO_METRICS_START_DATE'])
```

That would be one way of making this work. The reason is that `current_app` is bound to what is called the Flask Application Context, which is loaded after some other things have happened first.

So, I agree on fetching this date from configuration, I'm a bit hesitant in having this in such a "global" manner since in the future we might need to expose metrics for different time ranges (and possibly other parameters), in which case we would need a per-metric configuration or similar...

To not overthink things, let's just go with this approach now, and create an issue out of this comment to review in the future.